geode-core/src/main/java/org/apache/geode/internal/security/SecurityServiceFactory.java

```diff
@@ -51,6 +53,11 @@ public static SecurityService create(Properties securityProps, CacheConfig cache
 
   public static SecurityService create(Properties securityProps,
       SecurityManager preferredSecurityManager, PostProcessor preferredPostProcessor) {
+    String securityServiceClassName = System.getProperty(SECURITY_SERVICE_SYSTEM_PROPERTY);
```

Comment:
Is this system property get in a hot path? System property access is synchronized. This should probably be part of the static initializer for this class rather than on a per-call basis. System properties should not be mutated at runtime.

Reply:
It's not a hot path, security service is only created once when cache is created.